### ISSUE TITLE
False positive about interpolation quoting when joining lines with slash

### ISSUE THREAD

**Issue by e2** [rubocop/rubocop#1422]
False positive after upgrading to 0.27.0

Given a file:

``` ruby
FOO= "bar"\
  "baz"
```

I get (with a custom `.rubocop.yml`, but that's irrelevant):

```
a.rb:1:7: C: Prefer single-quoted strings inside interpolations.
FOO = "bar"\
      ^^^^^
a.rb:2:3: C: Prefer single-quoted strings inside interpolations.
  "baz"
  ^^^^^
```

But if I change it to single quotes, I get:

```
a.rb:1:7: C: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.
FOO = 'bar'\
      ^^^^^
a.rb:2:3: C: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.
  'baz'
  ^^^^^
```

so there seems to be no way to escape Rubocop's sense of justice...

Use case: I have VERY long lines of text and I'm trying to split them in a way
that Rubocop doesn't complain, and the only workaround I found is:

``` ruby
FOO =
  "bar" +
  "baz"
```

(Produces no offenses).
